src/Interpreters/JIT/CHJIT.cpp

```diff
@@ -483,15 +493,43 @@ std::string CHJIT::getMangledName(const std::string & name_to_mangle) const
     return mangled_name;
 }
 
-void CHJIT::runOptimizationPassesOnModule(llvm::Module & module) const
+#ifndef NDEBUG
+
+static std::string DumpModuleIR(const llvm::Module& module)
 {
+    std::string ir;
+    llvm::raw_string_ostream stream(ir);
+    module.print(stream, nullptr);
+    return ir;
+}
+
+#endif
+
+void CHJIT::runOptimizationPassesOnModule(llvm::Module & module, llvm::TargetIRAnalysis target_analysis) const
+{
+    /// llvm::DebugFlag = true; // comment out to enable debug output
+    const char *argv[] = {
+        "CH-JIT",
+        "-extra-vectorizer-passes", // Enable ExtraVectorizerPasses
+    };
+    int argc = sizeof(argv) / sizeof(argv[0]);
+    llvm::cl::ParseCommandLineOptions(argc, argv, "CH-JIT");
+
     llvm::LoopAnalysisManager lam;
     llvm::FunctionAnalysisManager fam;
     llvm::CGSCCAnalysisManager cgam;
     llvm::ModuleAnalysisManager mam;
-
+    fam.registerPass([&] { return target_analysis; });
```

Comment:
Strange order of lines. Do we actually need to do it?

Reply:
Not really.